bundles/org.openhab.binding.shelly/src/main/java/org/openhab/binding/shelly/internal/util/ShellyUtils.java

```diff
@@ -304,13 +304,17 @@ public static DateTimeType getTimestamp(String zone, long timestamp) {
             ZoneId zoneId = !zone.isEmpty() ? ZoneId.of(zone) : ZoneId.systemDefault();
             ZonedDateTime zdt = LocalDateTime.now().atZone(zoneId);
             int delta = zdt.getOffset().getTotalSeconds();
-            return new DateTimeType(Instant.ofEpochSecond(timestamp - delta));
+            return new DateTimeType(ZonedDateTime.ofInstant(Instant.ofEpochSecond(timestamp - delta), zoneId));
```

Comment:
This reverts a simplification from #17725, please revert:
```suggestion
            return new DateTimeType(Instant.ofEpochSecond(timestamp - delta));
```

See 03a0993b995e8f295e1521b063ae6be16c2e24cb.